packages/datacommons-mcp/datacommons_mcp/cli.py

```diff
@@ -13,46 +26,98 @@ def cli() -> None:
     logging.basicConfig(level=logging.INFO)
 
 
-@cli.group()
-def serve() -> None:
+def _validate_mode_options(ctx: Context, mode: str) -> None:
+    """Checks for options that are not applicable to the selected mode."""
+    provided_options = {
+        param.name
+        for param in ctx.command.params
+        if isinstance(param, Option)
+        and ctx.get_parameter_source(param.name) is not ParameterSource.DEFAULT
+    }
+    allowed_options = COMMON_OPTIONS.union(MODE_SPECIFIC_OPTIONS.get(mode, set()))
+    invalidly_used_options = provided_options - allowed_options
+
+    if invalidly_used_options:
+        # Format a user-friendly error message.
+        opts_str = ", ".join(f"--{opt}" for opt in sorted(invalidly_used_options))
+        raise click.UsageError(
+            f"The following option(s) are not applicable in '{mode}' mode: {opts_str}"
+        )
+
+
+def _run_api_key_validation(ctx: Context, *, skip_validation: bool) -> None:
+    """Runs the API key validation unless skipped."""
+    if not skip_validation:
+        try:
+            api_key = os.getenv("DC_API_KEY")
+            if not api_key:
+                raise InvalidAPIKeyError("DC_API_KEY is not set.")
+            validate_api_key(api_key)
+        except (InvalidAPIKeyError, APIKeyValidationError) as e:
+            click.echo(str(e), err=True)
+            click.echo(
+                "To obtain an API key, go to https://apikeys.datacommons.org and "
+                "request a key for the api.datacommons.org domain.",
+                err=True,
+            )
+            sys.stderr.flush()
+            ctx.exit(1)
+    else:
+        click.echo("Skipping API key validation as requested.", err=True)
```

Comment:
super nit: Should we write this to err only for stdio mode? Or is it okay for http as well?